### PR TITLE
chore: build image for digest and release on workflow_dispatch

### PR DESCRIPTION
**Pull request by stebenz**
### Definition of Ready

- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -72,24 +72,26 @@ jobs:
   container:
     needs: [compile]
     uses: ./.github/workflows/container.yml
+    secrets: inherit
     with:
-      image_name: 'ghcr.io/zitadel/zitadel'
+      build_image_name: 'ghcr.io/zitadel/zitadel-build'
```
Comment (livio-a) [inline .github/workflows/build.yml:77]: isn't there already a `build-cache` repository?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -72,24 +72,26 @@ jobs:
...
     with:
-      image: ${{ needs.container.outputs.image }}-debug
+      image: ${{ needs.container.outputs.build_image }}-debug
 
   release:
     uses: ./.github/workflows/release.yml
     needs: [version, core-unit-test, core-integration-test, lint, container, e2e]
     # TODO: trigger release on workflow_dispatch if: ${{ github.event_name == 'workflow_dispatch' }}
-    if: ${{ needs.version.outputs.published == 'true' }}
+    if: ${{ needs.version.outputs.published == 'true' && github.event_name == 'workflow_dispatch' }}
```
Comment (livio-a) [inline .github/workflows/build.yml:90]: please remove the TODO

File: .github/workflows/release-channels.yml [hunk 3]
```diff
@@ -35,11 +35,6 @@ jobs:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
-      - uses: tibdex/github-app-token@v1
-        id: generate-token
-        with:
-          app_id: ${{ secrets.APP_ID }}
-          private_key: ${{ secrets.APP_PRIVATE_KEY }}
```
Comment (livio-a) [inline .github/workflows/release-channels.yml:42]: is this change intended?

Comment (livio-a) [reply .github/workflows/release-channels.yml:42]: ah looks like it was not used anyway

**Comment by livio-a** [comment]
tried to test it on a fork and looks like the secrets are not passed correctly: 
image

https://github.com/zitadel/zitadel/actions/runs/5746648907/job/15576561849?pr=6311

File: .releaserc.js [hunk 4]
```diff
@@ -1,11 +1,24 @@
 module.exports = {
   branches: [
     { name: "next" },
-    { name: "next-rc", prerelease: "rc" }
+    { name: "next-rc", prerelease: "rc" },
+    { name: "pipeline-image", prerelease: "ignore-me" }
```
Comment (muhlemmer) [inline .releaserc.js:5]: should we remove this line before final merge of this PR?

**Review by muhlemmer — COMMENTED** [review]
Just a small comment.

Also, I'm currently trying a PR from a fork to see if it runs fine: https://github.com/zitadel/zitadel/pull/6344.

**Review by muhlemmer — APPROVED** [review]
